### PR TITLE
refactor(core): unify setPreviousOrParentTNode and setIsParent into single setPreviousOrParentTNode

### PR DESCRIPTION
**Pull request by mhevery**
…ingle setPreviousOrParentTNode

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: packages/platform-browser/src/dom/dom_renderer.ts [hunk 1]
```diff
@@ -156,8 +156,8 @@ class DefaultDomRenderer2 implements Renderer2 {
   nextSibling(node: any): any { return node.nextSibling; }
 
   setAttribute(el: any, name: string, value: string, namespace?: string): void {
-    if (namespace) {
-      name = `${namespace}:${name}`;
+    if (namespace !== undefined) {
```
Comment (pkozlowski-opensource) [inline packages/platform-browser/src/dom/dom_renderer.ts]: 👍 - in fact I've started a WIP PR with this (and similar changes to the renderer) today on my side.

"small PRs nit": I would argue that this should one go to a separate PR with more changes like this in the `dom_renderer.ts`

File: packages/platform-browser/src/dom/dom_renderer.ts [hunk 2]
```diff
@@ -156,8 +156,8 @@ class DefaultDomRenderer2 implements Renderer2 {
...
+      name = namespace + ':' + name;
```
Comment (pkozlowski-opensource) [inline packages/platform-browser/src/dom/dom_renderer.ts]: oh, interesting - I was assuming that the 2 would be equivalent, at least when transpiled to ES5 (https://www.typescriptlang.org/play/#src=%60%24%7B%22foo%22%7D%3A%24%7B%22bar%22%7D%60)

File: packages/core/src/render3/util/view_traversal_utils.ts [hunk 3]
```diff
@@ -49,7 +49,7 @@ export function getRootView(componentOrLView: LView | {}): LView {
  */
 export function findComponentView(lView: LView): LView {
   let rootTNode = lView[T_HOST];
-  while (rootTNode && rootTNode.type === TNodeType.View) {
+  while (rootTNode !== null && rootTNode.type === TNodeType.View) {
```
Comment (pkozlowski-opensource) [inline packages/core/src/render3/util/view_traversal_utils.ts]: 👍

File: packages/core/src/render3/instructions/text.ts [hunk 4]
```diff
@@ -10,12 +10,14 @@ import {TNodeType} from '../interfaces/node';
 import {RText, isProceduralRenderer} from '../interfaces/renderer';
 import {BINDING_INDEX, HEADER_OFFSET, RENDERER, TVIEW} from '../interfaces/view';
 import {appendChild, createTextNode} from '../node_manipulation';
-import {getLView, setIsParent} from '../state';
+import {getLView, setIsNotParent} from '../state';
 import {NO_CHANGE} from '../tokens';
 import {renderStringify} from '../util/misc_utils';
 import {getNativeByIndex} from '../util/view_utils';
+
```
Comment (pkozlowski-opensource) [inline packages/core/src/render3/instructions/text.ts]: Is it a formatter adding those new lines?

File: packages/core/src/render3/instructions/shared.ts [hunk 5]
```diff
@@ -1745,17 +1740,14 @@ export function storeBindingMetadata(lView: LView, prefix = '', suffix = ''): st
 
 export const CLEAN_PROMISE = _CLEAN_PROMISE;
 
-export function initializeTNodeInputs(tNode: TNode | null): PropertyAliases|null {
+export function initializeTNodeInputs(tNode: TNode): PropertyAliases|null {
```
Comment (pkozlowski-opensource) [inline packages/core/src/render3/instructions/shared.ts]: 👍 for this refactoring

File: packages/platform-browser/src/dom/dom_renderer.ts (hunk 2 above)
Comment (alfaproject) [reply packages/platform-browser/src/dom/dom_renderer.ts]: In ES2015+ target it's slower for the moment: https://jsperf.com/es6-string-literals-vs-string-concatenation

File: packages/core/src/render3/assert.ts [hunk 6]
```diff
@@ -36,8 +36,8 @@ export function assertPreviousIsParent(isParent: boolean) {
   assertEqual(isParent, true, 'previousOrParentTNode should be a parent');
 }
 
-export function assertHasParent(tNode: TNode) {
-  assertDefined(tNode.parent, 'previousOrParentTNode should have a parent');
+export function assertHasParent(tNode: TNode | null) {
+  assertDefined(tNode && tNode.parent, 'previousOrParentTNode should have a parent');
```
Comment (kara) [inline packages/core/src/render3/assert.ts]: Nit: It seems a bit confusing that we might print "previous node should have a parent" when the real problem might be that the previous node doesn't even exist. Seems like these should be separate checks, WDYT?

File: packages/platform-browser/src/dom/dom_renderer.ts (hunk 1 above)
Comment (kara) [reply packages/platform-browser/src/dom/dom_renderer.ts]: This change is what's causing CI to fail (I think namespace is sometimes `''` in View Engine), so +1 on moving it to a separate PR and debugging there.

File: packages/core/src/render3/instructions/shared.ts [hunk 7]
```diff
@@ -271,12 +271,16 @@ export function createNodeAtIndex(
     const tParentNode = parentInSameView ? parent as TElementNode | TContainerNode : null;
 
     tNode = tView.data[adjustedIndex] = createTNode(tParentNode, type, adjustedIndex, name, attrs);
+    if (index === 0) {
+      tView.firstChild = tNode;
+    }
```
Comment (kara) [inline packages/core/src/render3/instructions/shared.ts]: Ah, thanks for cleaning this up. IIRC, we originally didn't do this because view queries allocated slots, but they no longer do this, so checking index is better.

**Review by kara — APPROVED** [review]
LGTM, sans renderer change

File: packages/core/src/render3/assert.ts (hunk 6 above)
Comment (mhevery) [reply packages/core/src/render3/assert.ts]: fixed

File: packages/core/src/render3/instructions/shared.ts (hunk 7 above)
Comment (mhevery) [reply packages/core/src/render3/instructions/shared.ts]: 👍

File: packages/core/src/render3/instructions/text.ts (hunk 4 above)
Comment (mhevery) [reply packages/core/src/render3/instructions/text.ts]: reverted

File: packages/platform-browser/src/dom/dom_renderer.ts (hunk 1 above)
Comment (mhevery) [reply packages/platform-browser/src/dom/dom_renderer.ts]: I see, in that case I have to revert it.

**Comment by kara** [comment]
merge-assistance: global approval

**Comment by mhevery** [comment]
http://test/OCL:248432364:BASE:248432391:1557964271508:b22a26b6

**Comment by mhevery** [comment]
http://test/OCL:248432364:BASE:248603277:1558042972909:4edeadc9

**Comment by mhevery** [comment]
CARETAKER: The `ci-codefresh-bazel` build is stuck.

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM, thnx for doing all those changes / cleanups! 

Just one nit: slightly confused by doc formatting changes in https://github.com/angular/angular/pull/30453/commits/5651fd03bd59de6cafa10b8b60278362d1184a39 - did we have badly formatted code in master or is it your local formatter that is different?

**Review by kara — APPROVED** [review]
